logging/unit_tests/test__shutdown.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import mock
+
+
+class Test_setup_shutdown_stracktrace_reporting(unittest.TestCase):
+
+    def _call_fut(self, request):
+        from google.cloud.logging._shutdown import (
+            setup_stacktrace_crash_report)
+        return setup_stacktrace_crash_report(request)
+
+    def test_setup_shutdown_stacktrace_reporting_no_gae(self):
```

Comment:
1. You could simply name this `test_outside_of_gae`
1. You should actually mock this method to **make sure** that `_is_on_appengine()` returns `False`